### PR TITLE
Make dsconnect bindable to a specific IP

### LINKED ISSUE #3117: Limit dsconnect to IP
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:**
not related

**_Server Version_** (type `@revision` in game) **:**
ffc3a95

**_Source Branch_** (master/stable) **:**
master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:**
It's not possible to limit dsconnect to only listen to a specific ip address.

this would be of great help to test a modified version versus a vanilla version or to have different branches running or even to have a dev and a live game server on the same box at the same time

### PR DESCRIPTION
**Pull request by ymkc**
First attempt to fix #3117

Not sure it works under windows (cannot test it) and about the parameter names (as I just passed them through)

### CODE REVIEW

**Comment by teschnei** [comment]
can't merge until it compiles on Windows

if you don't know how to deal with the IP addresses, look at map.cpp (it reads the IP address as command line arg)

also, it's 4 spaces for a tab, not 8

File: conf/login_darkstar.conf [hunk 1]
```diff
@@ -32,6 +32,17 @@ stdout_with_ansisequence: no
 console_silent: 0
 
 #--------------------------------
+#TCP/IP parameters
+#--------------------------------
+
+uiLobbyDataIp: 0.0.0.0
```
Comment (TeoTwawki) [inline conf/login_darkstar.conf]: please default to 127.0.0.1 for IP

ui us? typical DSP user isn't going to know what your abbreviation is unless you explain it in comment above.

**Comment by ymkc** [comment]
Will try to fix the windows side, just have no way to test/compile it :|

As for the ui/us stuff, that's what was already there.
I can just drop them or change the them

**Comment by TeoTwawki** [comment]
point was on the settings end of it, ppl won't know what that means because they aren't familiar with the internal code you got them from. Can name the settings anything that looks reasonable that doesn't make you explain to much. :)

**Comment by teschnei** [comment]
It meant unsigned int/short, but the IP were changed to a string :)

**Comment by teschnei** [comment]
Also the default of 0.0.0.0 matches the old setting (INADDR_ANY)

**Comment by TeoTwawki** [comment]
make more sense to default to math the ip of other settings defaults though

**Comment by xdemolish** [comment]
remember to update xiloader to specify the port too

**Comment by ymkc** [comment]
Ok thanks for the input!

How about reducing/unifying the 3 IP params to just one? Does it make sense/is it reasonable/does it even work to have dsconnect running on 3 different IPs or even hosts?

**Comment by TeoTwawki** [comment]
> remember to update xiloader to specify the port too

feature already exists I thought, and just didn't' work because the lobby was still a static port

**Comment by xdemolish** [comment]
@teotwawki doesnt look like it
https://github.com/DarkstarProject/xiloader/blob/master/xiloader/network.cpp#L215
https://github.com/DarkstarProject/xiloader/blob/master/xiloader/network.cpp#L501

**Comment by TeoTwawki** [comment]
lol, so atom0s had a command line switch for it when he first released the thing, but it did nothing this entire time.

**Comment by ymkc** [comment]
- fixed the win32 build (missing header)
- set 127.0.0.1 as default IP
- used <creativemode> for var names

but can't for the life of baby jesus get that one tab from src/common/socket.cpp to be gone
it's not in my local copy and I even tried to remove/readd the whole line and commit it

File: src/common/socket.cpp [hunk 2]
```diff
@@ -827,7 +828,7 @@ int32 makeListenBind_tcp(uint32 ip, uint16 port,RecvFunc connect_client)
 	//set_nonblocking(fd, 1);
 
 	server_address.sin_family      = AF_INET;
-	server_address.sin_addr.s_addr = htonl(ip);
+        inet_pton(AF_INET, ip, &(server_address.sin_addr));
```
Comment (teschnei) [inline src/common/socket.cpp]: server_address.s_addr = inet_addr(ip);

this is what the map server does, and should be what you do unless it's not possible

Comment (ymkc) [inline src/common/socket.cpp]: @teschnei thanks, now i got you :)

File: src/common/socket.cpp [hunk 3]
```diff
@@ -827,7 +827,7 @@ int32 makeListenBind_tcp(uint32 ip, uint16 port,RecvFunc connect_client)
 	//set_nonblocking(fd, 1);
 
 	server_address.sin_family      = AF_INET;
-	server_address.sin_addr.s_addr = htonl(ip);
+        server_address.sin_addr.s_addr = inet_addr(ip);
```
Comment (teschnei) [inline src/common/socket.cpp]: there are 8 spaces on this indent, and there should be 4

Comment (ymkc) [inline src/common/socket.cpp]: resetted the file

**Comment by teschnei** [comment]
gonna do a quick test on windows when I get home then I'll merge it if it works as expected

**Pull request merged**